File: javascript/types/bools/automatic_casts_to_bool.js
#!/usr/bin/env node

let i = 0;
console.log("let i = 0; // (" + typeof i + ")");
if (i) console.log("casted to true");
else console.log("casted to false");

i = 1;
console.log("i = 1; // (" + typeof i + ")");
if (i) console.log("casted to true");
else console.log("casted to false");

i = 3;
console.log("i = 3; // (" + typeof i + ")");
if (i) console.log("casted to true");
else console.log("casted to false");

i = -3;
console.log("i = -3; // (" + typeof i + ")");
if (i) console.log("casted to true");
else console.log("casted to false");

console.log();

let f = 0.0;
console.log("let f = 0.0; // (" + typeof f + ")");
if (f) console.log("casted to true");
else console.log("casted to false");

f = 1.0;
console.log("f = 1.0; // (" + typeof f + ")");
if (f) console.log("casted to true");
else console.log("casted to false");

f = 3.0;
console.log("f = 3.0; // (" + typeof f + ")");
if (f) console.log("casted to true");
else console.log("casted to false");

f = -3.0;
console.log("f = -3.0; // (" + typeof f + ")");
if (f) console.log("casted to true");
else console.log("casted to false");

f = 0.1;
console.log("f = 0.1; // (" + typeof f + ")");
if (f) console.log("casted to true");
else console.log("casted to false");

f = 3.14;
console.log("f = 3.14; // (" + typeof f + ")");
if (f) console.log("casted to true");
else console.log("casted to false");

f = -3.14;
console.log("f = -3.14; // (" + typeof f + ")");
if (f) console.log("casted to true");
else console.log("casted to false");

console.log()

let s = "\0";
console.log("let s = \"\\0\"; // null (\\0) character (" + typeof s + ")");
if (s) console.log("casted to true");
else console.log("casted to false");

s = "";
console.log("s = \"\"; // empty string (" + typeof s + ")");
if (s) console.log("casted to true");
else console.log("casted to false");

s = " ";
console.log("s = \" \"; // space (" + typeof s + ")");
if (s) console.log("casted to true");
else console.log("casted to false");

s = "a";
console.log("s = \"a\"; // (" + typeof s + ")");
if (s) console.log("casted to true");
else console.log("casted to false");

s = "false";
console.log("s = \"false\"; // (" + typeof s + ")");
if (s) console.log("casted to true");
else console.log("casted to false");

s = "true";
console.log("s = \"true\"; // (" + typeof s + ")");
if (s) console.log("casted to true");
else console.log("casted to false");

console.log();

File: ruby/types/bools/automatic_casts_to_bool.rb
#!/usr/bin/ruby2.7

i = 0
puts "i = 0 # (#{i.class})"
if i then puts "casted to true"
else puts "casted to false" end

i = 1
puts "i = 1 # (#{i.class})"
if i then puts "casted to true"
else puts "casted to false" end

i = 3
puts "i = 3 # (#{i.class})"
if i then puts "casted to true"
else puts "casted to false" end

i = -3
puts "i = -3 # (#{i.class})"
if i then puts "casted to true"
else puts "casted to false" end

puts

f = 0.0
puts "f = 0.0 # (#{f.class})"
if f then puts "casted to true"
else puts "casted to false" end

f = 1.0
puts "f = 1.0 # (#{f.class})"
if f then puts "casted to true"
else puts "casted to false" end

f = 3.0
puts "f = 3.0 # (#{f.class})"
if f then puts "casted to true"
else puts "casted to false" end

f = -3.0
puts "f = -3.0 # (#{f.class})"
if f then puts "casted to true"
else puts "casted to false" end

f = 0.1
puts "f = 0.1 # (#{f.class})"
if f then puts "casted to true"
else puts "casted to false" end

f = 3.14
puts "f = 3.14 # (#{f.class})"
if f then puts "casted to true"
else puts "casted to false" end

f = -3.14
puts "f = -3.14 # (#{f.class})"
if f then puts "casted to true"
else puts "casted to false" end

puts

s = "\0"
puts "s = \"\\0\" # null (\\0) character (#{f.class})"
if s then puts "casted to true"
else puts "casted to false" end

s = ""
puts "s = \"\" # empty string (#{f.class})"
if s then puts "casted to true"
else puts "casted to false" end

s = " "
puts "s = \" \" # space (#{f.class})"
if s then puts "casted to true"
else puts "casted to false" end

s = "a"
puts "s = \"a\" # (#{f.class})"
if s then puts "casted to true"
else puts "casted to false" end

s = "false"
puts "s = \"false\" # (#{f.class})"
if s then puts "casted to true"
else puts "casted to false" end

s = "true"
puts "s = \"true\" # (#{f.class})"
if s then puts "casted to true"
else puts "casted to false" end

puts
